eksrollup/lib/aws.py

```diff
@@ -157,19 +157,30 @@ def modify_aws_autoscaling(asg_name, action):
     return response
 
 
-def scale_asg(asg_name, current_desired_capacity, new_desired_capacity, new_max_size):
+def scale_asg(asg_name, current_desired_capacity, new_desired_capacity, new_max_size, set_desired=True):
     """
     Changes the desired capacity of an asg
     """
-    logger.info('Setting asg desired capacity from {} to {} and max size to {}...'.format(current_desired_capacity, new_desired_capacity, new_max_size))
     if not app_config['DRY_RUN']:
-        response = client.update_auto_scaling_group(
-            AutoScalingGroupName=asg_name,
-            DesiredCapacity=new_desired_capacity,
-            MaxSize=new_max_size)
-        if response['ResponseMetadata']['HTTPStatusCode'] != requests.codes.ok:
-            logger.info('AWS scale up operation did not succeed. Exiting.')
-            raise Exception('AWS scale up operation did not succeed. Exiting.')
+        if set_desired:
```

Comment:
this block can be simplified to remove the duplication you can do something like 

```
 logger.info('Setting asg desired capacity from {} to {} and max size to {}...'.format(current_desired_capacity, new_desired_capacity, new_max_size))
response = []
if set_desired:
            response = client.update_auto_scaling_group(
                AutoScalingGroupName=asg_name,
                DesiredCapacity=new_desired_capacity,
                MaxSize=new_max_size)
else: 
            response = client.update_auto_scaling_group(
                AutoScalingGroupName=asg_name,
                MaxSize=new_max_size)

if response['ResponseMetadata']['HTTPStatusCode'] != requests.codes.ok:
                logger.info('AWS scale up operation did not succeed. Exiting.')
                raise Exception('AWS scale up operation did not succeed. Exiting.')


```